src/pages/LoginPage.jsx

```diff
@@ -0,0 +1,27 @@
+import React, { Component } from 'react';
+import { signin } from '../api/api';
+import { Redirect } from 'react-router-dom';
+import LoginForm from '../components/LoginForm';
+
+export default class LoginPage extends Component {
+  state = {
+    token: null,
+  };
+
+  signIn = async (loginData) => {
+    const [userDataError, userData] = await signin(loginData);
+    const token = userData.USERTOKEN || userData.ADMINTOKEN;
+    if (!userDataError && userData.USERTOKEN) {
+      localStorage.setItem('token', token);
+      this.setState({ token });
```

Comment:
why do you need to set token to state?

Reply:
I get the token in this component and set it to the state, and initially the token is null, below is a check if the token  received redirect to '/'